Feature/deepcopy add type alias support

epelc opened the pull request:
Adds support for deep copying type aliases.

Type aliases in go are the following form:

```
type MyOtherType struct {...}

type MyAlias = MyOtherType
```

Typically they are used for code migration but are also more prevalent with code using generics to make aliases for generic parameters that also have methods implemented. Named types do not inherit the implemented methods.

- Includes an additional change to print field names of unsupoprted field types.  This helps with large struct trees that might have a simple type like `any` embedded in them. Makes it much easier to figure where on the tree this any field is.

epelc committed:
add support for type aliases to deepcopy plugin

epelc committed:
make deepcopy plugin easier to debug unsupported fields
  - print field name in addition to field type(useful for type any or other base types on complicated struct trees)

awalterschulze commented:
This looks great

Could we add a test to the test/normal folder

Where an alias and DeepCopy is used?

Then I think it is ready to merge

epelc commented:
> This looks great
> 
> Could we add a test to the test/normal folder
> 
> Where an alias and DeepCopy is used?
> 
> Then I think it is ready to merge

Will do.

epelc committed:
add test for alias struct deep copy

epelc committed:
switch deep copy field copies from func to lexical scoping
  - speeds up copying by not calling extra inline funcs when only scoping is required

epelc committed:
handle type alias deep copy method detection

epelc commented:
@awalterschulze tests are added now. This should be good to go.

This also includes a change to switch from inline function calls `func() {// field assignments }()` to blocks`{// field assignments}` which has improved performance and is what I believe was actually intended. See [here](https://github.com/awalterschulze/goderive/pull/90/commits/e21280b6afb986c0fd1e5f0d2313cefb7af0ae01).

awalterschulze commented:
The code looks great.
So many great changes that could be applied to a lot of generators, but one generator at a time.

I am just a little worried about the testing, I would like another test that does deriveDeepCopy on the alias.
```go
this := &aliasToStruct{
		Level: 99,
	}
that := &aliasToStruct{}
deriveDeepCopyAliasToStruct(this, that)
// test equality
```

epelc committed:
add test for deriveDeepCopy in DeepCopy method defined on alias

epelc committed:
fix formating for all go files
run `go fmt ./...` on entire repo
  - many comments were misformatted. Makes contributing easier since modern go most editors auto run go fmt

epelc commented:
@awalterschulze added another test. It tests `dervieDeepCopy` defined as method on alias and also on underlying struct now(what I previously had).

awalterschulze commented:
Test looks great, thank you

Nit for future: please gofmt in a separate pull request, if it touches a lot of files you didn't touch.
But unpicking this seems painful now, so I'd rather merge the great work you did here.

Thanks you so much.
I think the scoping `{` instead of `func() {` can be applied to many generators.
I also think the type aliases could be better supported in many generators.

Also for solving #47 I think the agreed solution is https://github.com/awalterschulze/goderive/issues/47#issuecomment-998811227
But I think we can remove the command line argument factor and just go with 
https://github.com/awalterschulze/goderive/issues/47#issuecomment-1004061807
To be the only way to point to deriveTime via an alias.
